File: backend/src/main/java/com/devsuperior/movieflix/entities/Movie.java
package com.devsuperior.movieflix.entities;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@Data
@Entity
@Table(name = "tb_movie")
public class Movie implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String subTitle;
    private String year;
    private String imgUrl;
    @Column(columnDefinition = "TEXT")
    private String synopsis;
    @ManyToOne
    @JoinColumn(name = "genre_id")
    private Genre genre;
    @OneToMany(mappedBy = "movie")
    private List<Review> reviews = new ArrayList<>();
}

File: frontend/src/core/models/Movie.ts
import { Review } from "./Review";

export interface Movie {
    id: number;
    title: string;
    subTitle: string;
    year: string;
    imgUrl: string;
    synopsis: string;
    genreId: number;
    reviews: Review[];
}

export type MovieResponse = {
  content: Movie[];
  totalPages: number;
  number: number;
}